### PR TITLE
PRS ancestry dependent var adjustment 

### PR DESCRIPTION
**Pull request by kachulis**
Addresses #50 

We will want to wait until after discussions with eMerge prs wg before merging this change.

### CODE REVIEW

File: ImputationPipeline/ScoringPart.wdl [hunk 1]
```diff
@@ -304,47 +305,124 @@ task AdjustScores {
 
 			population_data = merge(population_pcs, population_scores, by.x="IID", by.y="X.IID")
 
-			# generate the linear model from the population data using the first 5 PCs
-			population_model = glm(SCORE1_SUM ~ PC1 + PC2 + PC3 + PC4 + PC5, data = population_data, family = "gaussian")
+			# generate the linear model from the population data using the first 4 PCs
+			population_model = glm(SCORE1_SUM ~ PC1 + PC2 + PC3 + PC4, data = population_data, family = "gaussian")
```
Comment (tmelman) [inline ImputationPipeline/ScoringPart.wdl:309]: On my future wishlist is not hardcoding number of pcs, but definitely not priority

Comment (kachulis) [reply ImputationPipeline/ScoringPart.wdl:309]: 👍

File: ImputationPipeline/ScoringPart.wdl [hunk 2]
```diff
@@ -304,47 +305,124 @@ task AdjustScores {
...
 
 			population_data$residual_score = resid(population_model)
+			population_data$residual_score2 = resid(population_model)^2
 			population_resid_mean = mean(population_data$residual_score)
```
Comment (tmelman) [inline ImputationPipeline/ScoringPart.wdl:313]: population_resid_mean and population_resid_sd are not used anywhere?

Comment (kachulis) [reply ImputationPipeline/ScoringPart.wdl:313]: good catch, no they are revenants from the old code.  will remove.

Comment (tmelman) [reply ImputationPipeline/ScoringPart.wdl:313]: same with population_data$residual_score (unsquared)

Comment (tmelman) [reply ImputationPipeline/ScoringPart.wdl:313]: same with population_data$residual_score

**Review by tmelman — APPROVED** [review]
A few minor cosmetic changes might be nice (removing variables declared but not used) but this is ok to merge.

**Comment by tmelman** [comment]
Just for posterity's sake: this method is essentially the same as what is called "Feasible Generalized Least Squares" done in PC space. It is an OLS/GLS extension for handling heteroskedasticity and learning the variances as well as the means. 

Here is a link to the results slides: 
https://docs.google.com/presentation/d/1YW5xwoa15lPvPtxbs0BihlFC2JUJ3XEBkVWEob14twY/edit#slide=id.ge706e21984_0_57

And here is one link for an R implementation of FGLS: 
https://www.rdocumentation.org/packages/RFGLS/versions/1.0/topics/fgls

**Pull request merged**